### ISSUE TITLE
Memory Leak in aireplay-ng

### ISSUE THREAD

**Issue by aircrack-ng** [aircrack-ng/aircrack-ng#1146]
**Reported by Moshe Kaplan  me on 14 Jul 2013 23:42 UTC**

Memory is allocated with `malloc`, but it is possible for the function to exit without storing a reference to the allocated memory or `free`ing it.

On aireplay-ng.c, line 3199-3202:
```
if ( opt.ringbuffer )
	arp = (struct ARP_req*) malloc( opt.ringbuffer * sizeof( struct ARP_req ) );
else
	arp = (struct ARP_req*) malloc( sizeof( struct ARP_req ) );

```

This function will likely need a fair amount of restructuring to prevent memory from leaking, since there are many exit points.

**Comment by aircrack-ng** [comment]
**Attachment by Moshe Kaplan  me on 16 Jul 2013 04:39 UTC**

Uploaded file [1146_patch.diff](https://github.com/aircrack-ng/trac-attachments/blob/master/1146/20130716-043926_771-1146_patch.diff)

**Comment by aircrack-ng** [comment]
**Comment by misterx on 14 Jul 2013 23:42 UTC**

Fixed in aircrack-ng/aircrack-ng@0fdf02b4e6a931bb6a8bd21baf3afcc801ed6c86
